src/OpenTelemetry/Trace/ActivitySourceAdapter.cs

```diff
@@ -101,8 +102,9 @@ private void RunGetRequestedData(Activity activity)
                 activity.Links);
 
             var samplingDecision = this.sampler.ShouldSample(samplingParameters);
-            activity.IsAllDataRequested = samplingDecision.IsSampled;
-            if (samplingDecision.IsSampled)
+            var isSampled = samplingDecision.Decision == SamplingDecision.Record || samplingDecision.Decision == SamplingDecision.RecordAndSampled;
```

Comment:
It seems `isSampled` is not very useful here?

Consider optimizing the flow for readability:

switch (samplingDecision.Decision)
{
case samplingDecision.Record:
  activity.IsAllDataRequested = true;
  break;
case SamplingDecision.RecordAndSampled:
  activity.IsAllDataRequested = true;
  activity.ActivityTraceFlags |= ActivityTraceFlags.Recorded;
  break;
}